### PR TITLE
mucs subtask2 baseline recipes (e2e and kaldi)

### PR DESCRIPTION
**Pull request by bloodraven66**
ToDo for subtask2  
- [x] add RESULTS.md
- [ ] add uploaded model info
- [x] add dataset info in egs/README.md
- [x] add stage -1, data download
- [x] add stage 0, data prep

### CODE REVIEW

**Comment by sw005320** [comment]
Thanks, @bloodraven66!
I have some quick comments.
- Please rename `mucs_subtask2/ASR_E2E`  to `mucs_subtask2/asr1`
- mucs or mucs21?
- Please remove `egs/mucs_subtask2/ASR_kaldi` and other unnecessary files

**Comment by bloodraven66** [comment]
> Thanks, @bloodraven66!
> I have some quick comments.
> 
> * Please rename `mucs_subtask2/ASR_E2E`  to `mucs_subtask2/asr1`
> * mucs or mucs21?
> * Please remove `egs/mucs_subtask2/ASR_kaldi` and other unnecessary files

@sw005320, I have done the changes.

**Comment by sw005320** [comment]
Cool!
Could you add your TODO items something like that?
- [ ] add `RESULTS.md`

So that we can easily monitor our progress and TODO items.

**Comment by bloodraven66** [comment]
Sure. I have edited the top comment.

**Comment by ftshijt** [comment]
Thanks for your effort! Besides the action items, could you please also read the recipe section at [CONTRIBUTING.md](https://github.com/espnet/espnet/blob/master/CONTRIBUTING.md)?

**Comment by bloodraven66** [comment]
@ftshijt I've updated the ToDos based on 1.3.1, 1.3.3 in CONTRIBUTING.md.

@sw005320 Should all the languages from the challenge be added at README.md? If so, is there a preference on how it should be displayed?

**Comment by sw005320** [comment]
> @sw005320 Should all the languages from the challenge be added at README.md? If so, is there a preference on how it should be displayed?

Yes, you can add all of them. Another way is to list the number of languages, but either is fine.

File: egs/mucs21_subtask2/asr1/run.sh [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    for x in train test ; do
+        steps/make_fbank_pitch.sh --cmd "$train_cmd" --nj ${nj} --write_utt2num_frames true \
+            data/${x} exp/make_fbank/${x} ${fbankdir}
+        utils/fix_data_dir.sh data/${x}
+    done
+
+
+    cp  -r  data/${train_set} data/${train_set}_org
+    cp  -r  data/${train_dev} data/${train_dev}_org
+    utils/perturb_data_dir_speed.sh 0.9  data/${train_set}_org  data/temp1
```
Comment (pengchengguo) [inline egs/mucs21_subtask2/asr1/run.sh:92]: Hi @bloodraven66,

Thanks so much for your efforts. A more simpler way for speed perturbation is to use `utils/speed_perturb.sh` like `egs/iwslt18/asr1/run.sh`. But it's not necessary. Besides, could you please change the soft link of `egs/mucs21_subtask2/asr1/steps` and `egs/mucs21_subtask2/asr1/utils` to `../../wsj/asr1/steps. These will same as other recipes.

**Comment by bloodraven66** [comment]
@pengchengguo I have changed it. 
I've added download_data, can you see if it can be improved?
I'll commit stage 0 in a few hours

**Comment by bloodraven66** [comment]
Hi, it got a bit delayed from my end due to server & VPN issues. I've added the stages. While Shreya shares RESULTS.ms and model link, I'll start with refactoring to egs2 on a branch.

**Comment by sw005320** [comment]
Cool.
I think we can merge this PR and we can have a follow-up PR to include RESULTS.ms and model link.
If you fix the CI error, I can merge this PR.

**Comment by sw005320** [comment]
https://github.com/espnet/espnet/pull/3362/checks?check_run_id=3234779754#step:7:141

File: egs/mucs21_subtask2/asr1/conf/decode.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+batchsize: 0
+beam-size: 10
+ctc-weight: 0.4
+lm-weight: 0.6
+maxlenratio: 0.0
+minlenratio: 0.0
+penalty: 0.0
+verbose: 8
```
Comment (sw005320) [inline egs/mucs21_subtask2/asr1/conf/decode.yaml:8]:
```suggestion
```

Comment (sw005320) [reply egs/mucs21_subtask2/asr1/conf/decode.yaml:8]: You can remove some unnecessary config parameters.

File: egs/mucs21_subtask2/asr1/local/download_data.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+#Copyright
+
```
Comment (sw005320) [inline egs/mucs21_subtask2/asr1/local/download_data.sh:4]: You can add copyright info with your name.

File: egs/mucs21_subtask2/asr1/local/gen_wavscp.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
```
Comment (sw005320) [inline egs/mucs21_subtask2/asr1/local/gen_wavscp.sh:1]: Could you briefly explain what this script does?

File: egs/mucs21_subtask2/asr1/local/prepare_data.sh [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+#local/prepare_data.sh data/hi-en/test/transcripts/wav.scp data/hi-en/test/audio/ out.scp
```
Comment (sw005320) [inline egs/mucs21_subtask2/asr1/local/prepare_data.sh:3]: ditto
Could you briefly explain what this script does?

**Review by sw005320 — COMMENTED** [review]
LGTM
I have some minor comments.

File: egs/mucs21_subtask2/asr1/local/gen_wavscp.sh (hunk 4 above)
Comment (bloodraven66) [reply egs/mucs21_subtask2/asr1/local/gen_wavscp.sh:1]: > Cool.
> I think we can merge this PR and we can have a follow-up PR to include RESULTS.ms and model link.
> If you fix the CI error, I can merge this PR.

Okay Sure. Will do it in the morning.

File: egs/mucs21_subtask2/asr1/local/prepare_data.sh (hunk 5 above)
Comment (bloodraven66) [reply egs/mucs21_subtask2/asr1/local/prepare_data.sh:3]: This script replicates gen_wavscp functionality(I can remove gen_wavscp). It is necessary to change the paths present in the pipeline Shreya has for 2 datasets.

**Comment by sw005320** [comment]
@bloodraven66, I just merged this PR.
Thanks a lot!
Please continue to work on the rest of the items.

**Comment by bloodraven66** [comment]
> @bloodraven66, I just merged this PR.
> Thanks a lot!
> Please continue to work on the rest of the items.

Great. The action items are porting to espnet2 and the same steps for subtask1 correct? Do you have anything else in mind?

**Comment by sw005320** [comment]
Yes!!
And I'm writing some more possible actions elated to the tutorial by email.

**Pull request merged**